### PR TITLE
Feature/test cases

### PR DESCRIPTION
**Pull request by adnan1naeem**
PluginSettings App tests added

### CODE REVIEW

**Comment by adnan1naeem** [comment]
@artalat  kindly enable codacy

File: package.json [hunk 1]
```diff
@@ -76,18 +77,36 @@
     "node": ">=8.9"
   },
   "dependencies": {
+    "@babel/runtime": "^7.4.3",
```
Comment (artalat) [inline package.json]: please remove extra deps

File: src/HomeScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { ScrollView, StatusBar, View } from 'react-native';
+import { Button } from '@bluebase/components';
+import React from 'react';
+// tslint:disable: jsx-no-lambda max-line-length
+
+export class HomeScreen extends React.Component<any> {
```
Comment (artalat) [inline src/HomeScreen.tsx]: what is the purpose of this component?

File: src/components/AboutSettingsList/AboutSettingsList.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+			const version = BB.Configs.getValue('version');
+
+			return version && <List.Item title={__('Version')} description={__(version)} />;
+		}
+
+			renderAuthor = ({ __ }: IntlContextData) => {
+			const BB: BlueBase = this.context;
+			const { styles = {} } = this.props;
+
+			const author = BB.Configs.getValue('author');
```
Comment (artalat) [inline src/components/AboutSettingsList/AboutSettingsList.tsx]: can we add a default author in configs?

File: src/layouts/SettingsPageItem/SettingsPageItemDesktop.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+export interface SettingsPageItemDesktopProps extends SettingsPageItemProps {
+	styles?: Partial<SettingsPageItemDesktopStyles>;
+}
+
+export class SettingsPageItemDesktop extends React.PureComponent<SettingsPageItemDesktopProps> {
+			static defaultProps: Partial<SettingsPageItemDesktopProps> = {
+		// isMobile: true,
+		};
+
+			private Component =
```
Comment (artalat) [inline src/layouts/SettingsPageItem/SettingsPageItemDesktop.tsx]: please fix indentation

File: src/layouts/SettingsPageItem/SettingsPageItemMobile.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+export class SettingsPageItemMobile extends React.PureComponent<SettingsPageItemMobileProps> {
+			static defaultProps: Partial<SettingsPageItemMobileProps> = {};
+
+			private Component =
+		this.props.component && typeof this.props.component === 'string'
+			? getComponent(this.props.component)
+			: this.props.component;
+
+			static defaultStyles = (theme: Theme): SettingsPageItemMobileStyles => ({
```
Comment (artalat) [inline src/layouts/SettingsPageItem/SettingsPageItemMobile.tsx]: please fix indentation

File: .releaserc [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{
```
[the remark this answers is not in the document]
Comment (adnan1naeem) [reply .releaserc]: follow ,releaserc of master branch https://github.com/BlueBaseJS/plugin-launcher/blob/master/.releaserc @artalat

File: src/HomeScreen.tsx (hunk 2 above)
Comment (adnan1naeem) [reply src/HomeScreen.tsx]: i just removed the component as it was not helpful

File: src/components/AboutSettingsList/AboutSettingsList.tsx (hunk 3 above)
Comment (adnan1naeem) [reply src/components/AboutSettingsList/AboutSettingsList.tsx]: yes we can  but what who would be the default author and why we want to do this

File: .releaserc (hunk 6 above)
[the remark this answers is not in the document]
Comment (adnan1naeem) [reply .releaserc]: please tell me what needs to be done here @artalat

File: package.json [hunk 7]
```diff
@@ -76,18 +77,35 @@
     "node": ">=8.9"
   },
   "dependencies": {
+    "@bluebase/components": "^5.10.0",
+    "compression": "^1.7.4",
```
Comment (artalat) [inline package.json]:
```suggestion
```

File: package.json [hunk 8]
```diff
@@ -76,18 +77,35 @@
...
     "deepmerge": "^3.2.0",
+    "expo": "^32.0.0",
+    "metro-react-native-babel-preset": "^0.53.1",
```
Comment (artalat) [inline package.json]:
```suggestion
```

File: package.json [hunk 9]
```diff
@@ -76,18 +77,35 @@
...
     "react-native": "^0.57.8",
-    "tslib": "^1.9.3"
+    "source-map-support": "^0.5.12",
```
Comment (artalat) [inline package.json]:
```suggestion
```

File: package.json [hunk 10]
```diff
@@ -76,18 +77,35 @@
...
+    "tslib": "^1.9.3",
+    "uuid": "^3.3.2"
```
Comment (artalat) [inline package.json]:
```suggestion
```

File: package.json [hunk 11]
```diff
@@ -76,18 +77,35 @@
...
   "dependencies": {
+    "@bluebase/components": "^5.10.0",
+    "compression": "^1.7.4",
     "deepmerge": "^3.2.0",
+    "expo": "^32.0.0",
+    "metro-react-native-babel-preset": "^0.53.1",
     "react-native": "^0.57.8",
-    "tslib": "^1.9.3"
+    "source-map-support": "^0.5.12",
+    "tslib": "^1.9.3",
```
Comment (artalat) [inline package.json]:
```suggestion
    "tslib": "^1.9.3"
```

File: .releaserc (hunk 6 above)
[the remark this answers is not in the document]
Comment (artalat) [reply .releaserc]: @adnan1naeem you need to extend the .relesaerc file from code-standards

**Comment by artalat** [comment]
@adnan1naeem when can we close this?

**Comment by adnan1naeem** [comment]
> @adnan1naeem when can we close this?

please merge it all has been resolved

**Pull request merged**